nep-xx.mediawiki

```diff
@@ -0,0 +1,102 @@
+<pre>
+  NEP:
+  Title: Contract Basic Methods Guideline
+  Author: Owen Zhang <zhangzhihong@ngd.neo.org>, Fernando Díaz Toledano <shargon@gmail.com>, Erik Zhang <erik@neo.org>
+  Type: Informational
+  Status: Draft
+  Created: 2021-04-19
+  Replaces: 0
+</pre>
+
+
+
+
+
+
+==Abstract==
+
+This proposal outlines a basic method guideline for the NEO blockchain that will provide systems with a generalized interaction mechanism for smart contract initial deploy, update and destroy.
+
+==Motivation==
+
+As the NEO blockchain scales, Smart Contract Initial Deploy, Update and Destroy will become increasingly important.  Without guideline for <code>update</code> and <code>destroy</code> method, systems will be required to maintain a unique API for each contract, regardless of their similarity to other contracts which makes application development very inconvenient. This guideline can lead most contracts to implement the same basic methods with the same types of parameters but still accept diversity.
+
+==Specification==
+
+In the method definitions below, we provide both the definitions of the functions as they are defined in the contract as well as the invoke parameters.
+
+===Methods===
+
+====_deploy====
+
+<pre>
+{
+  "name": "_deploy",
+  "safe": false,
+  "parameters": [
+    {
+      "name": "data",
+      "type": "Any"
+    },
+    {
+      "name": "update",
+      "type": "Boolean"
+    }
+  ],
+  "returntype": "Void"
+}
+</pre>
+
+This is an optional function which will be automatically executed by ContractManagement contract when a contract is first deployed or updated.
+
+The parameter <code>data</code> can be any type of supported parameters for contract-specific purpose. It's the same <code>data</code> that is accepted by Deploy() and Update() in ContractManagement contract.
+
+<code>update</code> is true when contract is updated via <code>update()<code> and it's false on initial deployment via <code>_deploy()<code>.
+
+====update====
+
+<pre>
```

Comment:
In the view https://github.com/neo-project/proposals/pull/154/files?short_path=6caa849#diff-6caa8493f36ba42782e59054f66c45a43f6c170927ce9383eccda13535890ea1 this region is wrong displayed

Reply:
Fixed.